src/index.ts

```diff
@@ -0,0 +1,595 @@
+      },
+    });
+
+    parser.add_argument("--integrationType", {
+      help: "Integration Type - Intended for internal use only.",
+      required: false,
+      default: CONSTANTS.SOOS.DEFAULT_INTEGRATION_TYPE,
+    });
+
+    parser.add_argument("--logLevel", {
+      help: "Minimum level to show logs: PASS, IGNORE, INFO, WARN or FAIL.",
+      default: LogLevel.INFO,
+      required: false,
+      type: (value: string) => {
+        return ensureEnumValue(LogLevel, value);
+      },
+    });
+
+    parser.add_argument("--onFailure", {
+      help: "Action to perform when the scan fails. Options: fail_the_build, continue_on_failure.",
+      default: OnFailure.Continue,
+      required: false,
+      type: (value: string) => {
+        return ensureEnumValue(OnFailure, value);
+      },
+    });
+
+    parser.add_argument("--operatingEnvironment", {
+      help: "Set Operating environment for information purposes only.",
+      required: false,
+    });
+
+    parser.add_argument("--outputFormat", {
+      help: "Output format for vulnerabilities: only the value SARIF is available at the moment",
+      required: false,
+      type: (value: string) => {
+        return ensureEnumValue(OutputFormat, value);
+      },
+    });
+
+    parser.add_argument("--packageManagers", {
+      help: "A list of package managers, delimited by comma, to include when searching for manifest files.",
+      required: false,
+      default: [],
+      type: (value: string) => {
+        const values = value.split(",");
+        values.map((value) => {
+          return ensureEnumValue(PackageManagerType, value);
+        });
+        return values;
+      },
+    });
+
+    parser.add_argument("--projectName", {
+      help: "Project Name - this is what will be displayed in the SOOS app.",
+      required: true,
+      type: (value: string) => {
+        return ensureNonEmptyValue(value, "projectName");
+      },
+    });
+
+    parser.add_argument("--scriptVersion", {
+      required: false,
+    });
+
+    parser.add_argument("--sourceCodePath", {
+      help: "Root path to begin recursive search for manifests.",
+      default: process.cwd(),
+      required: false,
+    });
+
+    parser.add_argument("--verbose", {
+      help: "Enable verbose logging.",
+      action: "store_true",
+      default: false,
+      required: false,
+    });
+
+    parser.add_argument("--workingDirectory", {
+      help: "Absolute path where SOOS may write and read persistent files for the given build. eg Correct: /tmp/workspace/ | Incorrect: ./bin/start/",
+      default: process.cwd(),
+      required: false,
+    });
+
+    soosLogger.info("Parsing arguments");
+    return parser.parse_args();
+  }
+
+  async runAnalysis(): Promise<void> {
+    let projectHash: string | undefined;
+    let branchHash: string | undefined;
+    let analysisId: string | undefined;
+
+    const soosAnalysisApiClient = new SOOSAnalysisApiClient(this.args.apiKey, this.args.apiURL);
```

Comment:
need to instantiate `soosAnalysisApiClient` because we need to do `getSupportedManifests` that doesnt make sense to be on `AnalysisService`

Reply:
(e.g. getScaSupportedManifests getSbomSupportedFormats)

Reply:
another option is to expose the underlying ApiClients on the service as properties, so we can call specific methods there.